### PR TITLE
Refine colreduction fusion strategy of kStitch.

### PR DESCRIPTION
**Pull request by yunzhongOvO**
This PR will add more strict fusion rule for fusionType `kColReduction` : guarantee that the OPs to be fused are both Rank2ColReduction and have same shapes.

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc [hunk 1]
```diff
@@ -650,7 +650,8 @@ LogicalResult lowerWithScheduleColReduction(
 
   SmallVector<Value, 4> yield_values_for_if;
 
-  ValueRange load_index({row_index, col_index});
+  SmallVector<Value, 2> multidim_load_index({row_index, col_index});
+  ValueRange load_index(multidim_load_index);
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc:654]: `SmallVector` could be cast `ValueRange` implicitly i think ...

Comment (yunzhongOvO) [reply tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc:654]: yes it is~

**Review by qiuxiafei — DISMISSED** [review]
LGTM~ pls clear the tests and merge.

File: .github/workflows/pytorch_pre_cpu.yml
[the remark this answers is not in the document]
Comment (yunzhongOvO) [reply .github/workflows/pytorch_pre_cpu.yml:1]: The version of PyTorch of pre CI tasks is outdated and the task will fail, and the available versions are always changing.

File: tao_compiler/mlir/disc/transforms/fusion_utils_stitch_gpu.cc [hunk 2]
```diff
@@ -62,13 +62,27 @@ bool StitchGpuFusionStrategy::tryFuse(ShapeAnalysis& shapeAnalysis,
...
+  bool has_col_reduction = llvm::any_of(target.getOpList(), [](Operation* op) {
     return isRank2ColReduction(op);
   });
 
-  if (has_row_reduction && has_col_reduciton) {
+  if (has_row_reduction && has_col_reduction) {
     return false;
   }
+
+  if (has_col_reduction) {
```
[the remark this answers is not in the document]
Comment (yunzhongOvO) [reply tao_compiler/mlir/disc/transforms/fusion_utils_stitch_gpu.cc:73]: A shape constraint is also set in `initFusionPattern` later, here I will delete the same shape constraint.

File: tao_compiler/mlir/disc/transforms/fusion_utils.cc [hunk 3]
```diff
@@ -1478,13 +1478,27 @@ bool BaseGpuFusionStrategy::tryFuse(ShapeAnalysis& shapeAnalysis,
   bool has_row_reduction = llvm::any_of(target.getOpList(), [](Operation* op) {
     return isRank2RowReduction(op);
   });
-  bool has_col_reduciton = llvm::any_of(target.getOpList(), [](Operation* op) {
+  bool has_col_reduction = llvm::any_of(target.getOpList(), [](Operation* op) {
```
[the remark this answers is not in the document]
Comment (yunzhongOvO) [reply tao_compiler/mlir/disc/transforms/fusion_utils.cc]: Done.

**Review by Yancey1989 — APPROVED** [review]
LGTM!

**Pull request merged**